handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslClientContext.java

```diff
@@ -0,0 +1,288 @@
+                    if (keyCertChain != null && key != null) {
+                        setKeyMaterial(ctx, keyCertChain, key, keyPassword);
+                    }
+                } else {
+                    if (keyCertChain != null) {
+                        keyManagerFactory = buildKeyManagerFactory(
+                                keyCertChain, key, keyPassword, keyManagerFactory);
+                    }
+                    if (keyManagerFactory != null) {
+                        X509KeyManager keyManager = chooseX509KeyManager(keyManagerFactory.getKeyManagers());
+                        OpenSslKeyMaterialManager materialManager = useExtendedKeyManager(keyManager) ?
+                                new OpenSslExtendedKeyMaterialManager(
+                                        (X509ExtendedKeyManager) keyManager, keyPassword) :
+                                new OpenSslKeyMaterialManager(keyManager, keyPassword);
+                        SSLContext.setCertRequestedCallback(ctx, new OpenSslCertificateRequestedCallback(
+                                engineMap, materialManager));
+                    }
+                }
+            } catch (Exception e) {
+                throw new SSLException("failed to set certificate and key", e);
+            }
+
+            SSLContext.setVerify(ctx, SSL.SSL_VERIFY_NONE, VERIFY_DEPTH);
+
+            try {
+                if (trustCertCollection != null) {
+                    trustManagerFactory = buildTrustManagerFactory(trustCertCollection, trustManagerFactory);
+                } else if (trustManagerFactory == null) {
+                    trustManagerFactory = TrustManagerFactory.getInstance(
+                            TrustManagerFactory.getDefaultAlgorithm());
+                    trustManagerFactory.init((KeyStore) null);
+                }
+                final X509TrustManager manager = chooseTrustManager(trustManagerFactory.getTrustManagers());
+
+                // IMPORTANT: The callbacks set for verification must be static to prevent memory leak as
+                //            otherwise the context can never be collected. This is because the JNI code holds
+                //            a global reference to the callbacks.
+                //
+                //            See https://github.com/netty/netty/issues/5372
+
+                // Use this to prevent an error when running on java < 7
+                if (useExtendedTrustManager(manager)) {
+                    SSLContext.setCertVerifyCallback(ctx,
+                            new ExtendedTrustManagerVerifyCallback(engineMap, (X509ExtendedTrustManager) manager));
+                } else {
+                    SSLContext.setCertVerifyCallback(ctx, new TrustManagerVerifyCallback(engineMap, manager));
+                }
+            } catch (Exception e) {
+                throw new SSLException("unable to setup trustmanager", e);
+            }
+        }
+        return new OpenSslClientSessionContext(thiz);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)